library/Asana/Api/Task.hs

```diff
@@ -184,3 +187,17 @@ putCustomFields taskId fields = put ("/tasks/" <> T.unpack (gidToText taskId))
 
 taskUrl :: Task -> Text
 taskUrl Task {..} = "https://app.asana.com/0/0/" <> gidToText tGid <> "/f"
+
+extractNumberField :: Text -> Task -> Maybe CustomField
+extractNumberField fieldName Task {..} =
+  headMay $ flip mapMaybe tCustomFields $ \case
+    customField@(CustomNumber _ t _) ->
+      if t == fieldName then Just customField else Nothing
+    _ -> Nothing
+
+extractEnumField :: Text -> Task -> Maybe CustomField
+extractEnumField fieldName Task {..} =
+  headMay $ flip mapMaybe tCustomFields $ \case
```

Comment:
Also `listToMaybe` is in `Data.Maybe` if you don't want to add `safe` for `headMay`